client/lib/signup/flow-controller.js

```diff
@@ -130,6 +130,45 @@ assign( SignupFlowController.prototype, {
 		}.bind( this ) );
 	},
 
+	/**
+	 * This assertion helper validates that there are dependencies for the step provided
+	 * by completed steps
+	 * @param step The step to validate
+	 */
+	_assertStepDependencies( step ) {
+		const dependencies = steps[ step.stepName ].dependencies || [];
+		const dependenciesFound = pick( SignupDependencyStore.get(), dependencies );
+		const dependenciesSatisfied = dependencies.length === keys( dependenciesFound ).length;
+		const signupProgress = filter(
+			SignupProgressStore.get(),
+			_step => ( -1 !== this._flow.steps.indexOf( _step.stepName ) )
+		);
+
```

Comment:
I debated with myself about it, both this method and `_canProcessStep` require those.

Another solution could be remove all of this here and pass them from `_canProcessStep` into this method, but that seems like a lot of parameters to me, so I chose having only the `step` as explicit parameter and deduce everything else, this have the advantage of being more self-contained method